### PR TITLE
add optional tokio-console feature

### PR DESCRIPTION
**Pull request by PSeitz**
add optional tokio-console feature

### CODE REVIEW

**Comment by fulmicoton** [comment]
Cool! Can you show us tomorrow?

**Comment by PSeitz** [comment]
yes, but making it work with CI requires some changes

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -39,6 +39,11 @@ The process is simple and fast. Upon your first pull request, you will be prompt
 2. Start the Jaeger services (UI, collector, agent, ...) running the command `make docker-compose-up DOCKER_SERVICES=jaeger`
 3. Open your browser and visit [localhost:16686](http://localhost:16686/)
 
+## Using tokio console
+1. Clone [tokio-console repository](https://github.com/tokio-rs/console) next to your quickwit repo
```
Comment (fulmicoton) [inline CONTRIBUTING.md:43]:
```suggestion
1. Install tokio-console by running `cargo install tokio-console`.
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -39,6 +39,11 @@ The process is simple and fast. Upon your first pull request, you will be prompt
...
+2. Build the binary with tokio unstable flag `RUSTFLAGS="--cfg tokio_unstable" cargo build`
+3. Launch a long running command such as index and activate tokio with the: `QUICKWIT_TOKIO_CONSOLE_ENABLED=1 target/debug/quickwit index ...`
+4. Run the console in the tokio-console repo `cargo run`
```
Comment (fulmicoton) [inline CONTRIBUTING.md:46]:
```suggestion
4. Run `tokio-console`.
```

File: quickwit-actors/src/async_actor.rs [hunk 3]
```diff
@@ -90,15 +90,35 @@ pub(crate) fn spawn_async_actor<A: AsyncActor>(
...
     .instrument(span);
-    let join_handle: JoinHandle<()> = tokio::spawn(loop_async_actor_future);
+
+    let join_handle: JoinHandle<()> = spawn_named(loop_async_actor_future, &actor_instance_id);
+
     ActorHandle::new(state_rx, join_handle, ctx_clone, exit_status_rx)
 }
 
+#[track_caller]
+pub(crate) fn spawn_named<T>(
```
Comment (fulmicoton) [inline quickwit-actors/src/async_actor.rs:106]:
```suggestion
fn spawn_named<T>(
```

File: quickwit-actors/src/async_actor.rs [hunk 4]
```diff
@@ -90,15 +90,35 @@ pub(crate) fn spawn_async_actor<A: AsyncActor>(
     let ctx_clone = ctx.clone();
     let span = actor.span(&ctx_clone);
     let (exit_status_tx, exit_status_rx) = watch::channel(None);
+    let actor_instance_id = ctx.actor_instance_id().to_string();
```
Comment (fulmicoton) [inline quickwit-actors/src/async_actor.rs:93]: is the to_string() necessary?

File: quickwit-actors/src/sync_actor.rs [hunk 5]
```diff
@@ -72,22 +72,37 @@ pub(crate) fn spawn_sync_actor<A: SyncActor>(
...
+            let exit_status = sync_actor_loop(actor, inbox, ctx, state_tx);
+            let _ = exit_status_tx.send(Some(exit_status));
+        },
+        &actor_instance_id,
+    );
     ActorHandle::new(state_rx, join_handle, ctx_clone, exit_status_rx)
 }
 
+#[track_caller]
+pub(crate) fn spawn_named_blocking<F, R>(f: F, _name: &str) -> tokio::task::JoinHandle<R>
```
Comment (fulmicoton) [inline quickwit-actors/src/sync_actor.rs:92]:
```suggestion
fn spawn_named_blocking<F, R>(f: F, _name: &str) -> tokio::task::JoinHandle<R>
```

File: quickwit-cli/src/main.rs [hunk 6]
```diff
@@ -43,6 +43,13 @@ fn event_format(
 }
 
 fn setup_logging_and_tracing(level: Level) -> anyhow::Result<()> {
+    #[cfg(feature = "tokio-console")]
+    {
+        if std::env::var_os(QUICKWIT_TOKIO_CONSOLE_ENABLED_ENV_KEY).is_some() {
+            console_subscriber::init();
+            return Ok(());
```
Comment (fulmicoton) [inline quickwit-cli/src/main.rs:50]:
```suggestion
```

**Review by fulmicoton — CHANGES_REQUESTED** [review]
Not sure if it requires changes or not.

See comments inline

File: quickwit-actors/src/async_actor.rs (hunk 4 above)
Comment (PSeitz) [reply quickwit-actors/src/async_actor.rs:93]: yes because the context is moved

File: quickwit-cli/src/main.rs (hunk 6 above)
Comment (PSeitz) [reply quickwit-cli/src/main.rs:50]: yes it needs to be disabled

**Review by fulmicoton — APPROVED** [review]

**Pull request merged**